File: app/src/main/java/me/rei_m/hyakuninisshu/domain/model/quiz/KarutaExam.java
package me.rei_m.hyakuninisshu.domain.model.quiz;

import android.support.annotation.NonNull;

import me.rei_m.hyakuninisshu.domain.AbstractEntity;

public class KarutaExam extends AbstractEntity<KarutaExam, KarutaExamIdentifier> {

    public static final int MAX_HISTORY_COUNT = 10;

    private final KarutaExamResult result;

    public KarutaExam(@NonNull KarutaExamIdentifier identifier,
                      @NonNull KarutaExamResult karutaExamResult) {
        super(identifier);
        this.result = karutaExamResult;
    }

    public KarutaExamResult result() {
        return result;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;

        KarutaExam that = (KarutaExam) o;

        return result.equals(that.result);

    }

    @Override
    public int hashCode() {
        int result1 = super.hashCode();
        result1 = 31 * result1 + result.hashCode();
        return result1;
    }

    @Override
    public String toString() {
        return "KarutaExam{" +
                "result=" + result +
                "} " + super.toString();
    }
}

File: domain/src/main/java/me/rei_m/hyakuninisshu/domain/model/quiz/KarutaExam.kt
/*
 * Copyright (c) 2018. Rei Matsushita
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
 * the License for the specific language governing permissions and limitations under the License.
 */

/* ktlint-disable package-name */
package me.rei_m.hyakuninisshu.domain.model.quiz

import me.rei_m.hyakuninisshu.domain.AbstractEntity
import java.util.*

/**
 * 百人一首の力試し.
 */
class KarutaExam(
    identifier: KarutaExamIdentifier,
    val tookDate: Date,
    val result: KarutaExamResult
) : AbstractEntity<KarutaExamIdentifier>(identifier) {

    override fun toString() = "KarutaExam(tookDate=$tookDate, result=$result)"

    companion object {
        const val MAX_HISTORY_COUNT = 10
    }
}
